docs/general-concepts/javascript/adding-javascript.md

```diff
@@ -0,0 +1,270 @@
+Note that you can use minified JavaScript, as in the "myjs2" example.
+If you include both myjs2.js and myjs2.min.js in the js folder, 
+then if you switch on Joomla Debug in the Global Configuration the Web Asset Manager will include the non-minified version. 
+
+**attributes** - these are attributes which just get mapped to attributes of the HTML `<script>` tag within the page HTML output.
+
+**dependencies** - the name or names of other assets upon which your code depends, for example in the json above:
+- "com_example.myjs1" is dependent upon Joomla "core" (specified in media/system/joomla.asset.json as the Joomla core.js library)
+and upon "jquery" (specified in media/vendor/joomla.asset.json)
+- "com_example.myjs2" is dependent upon "form.validate" (specified in media/system/joomla.asset.json, for performing form field validation)
+- "com_example.myjs2" is also dependent upon "com_example.myjs1" (in the same joomla.asset.json file),
+so when myjs2.js is loaded Joomla will also load myjs1.js, and all of the dependencies of "com_example.myjs1" as well. 
+
+The Web Asset Manager orders the `<script>` tags in the HTML to ensure that the browser processes the dependencies first.
+
+How do you know the names of the Joomla libraries to include as dependencies? 
+Entries in the [Joomla JavaScript Library](./js-library/index.md) may help, but you may need to look through the code files
+- Joomla library assets are in media/system/joomla.asset.json, and refer to code in media/system/js
+- vendor library assets are in media/vendor/joomla.asset.json, and refer to code in media/vendor/js
+
+These two joomla.asset.json files are always processed by Joomla, so their assets are always available.
+
+**version** - the version you want to apply to your JavaScript code. 
+When you update your code then if you increase the version number then Joomla will force the browser to reload your code, instead of using a cached copy.
+(It does this by adding your version as a query parameter in the URL of your js file).
+
+You also need to get Joomla to copy your joomla.asset.json file into the media folder:
+
+```xml title="Extension manifest file"
+    <media destination="com_example" folder="media">
+        <filename>joomla.asset.json</filename>
+        <folder>js</folder>
+    </media>
+```
+
+### 3 Get Joomla to process your joomla.asset.json file
+
+If your extension is a component or a template then Joomla performs this step automatically for you, and you have nothing to do.
+
+If your extension is a plugin or a module then you need to register your file with the Web Asset Manager in your PHP code:
+
+```php
+$app = Factory::getApplication();
+$document = $app->getDocument();
+$wa = $document->getWebAssetManager();
+$wa->getRegistry()->addExtensionRegistryFile('mod_example');
+```
+
+The code above will get Joomla to process media/mod_example/joomla.asset.json.
+
+### 4 Use your asset in your PHP Code
+
+```php
+$app = Factory::getApplication();
```

Comment:
Normally you use the webasset in context of the View or the template override it self, both cases `$this->getDocument()` is the better way.

Reply:
So a page "how to get object XYZ" could make sense, to explain it in an "easy" way and not need to explain the full concept.